### PR TITLE
[Filters] Add semantic markup to header

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### WHY are these changes introduced?

Fixes - internal issue linked below

### WHAT is this pull request doing?

* Adding semantic HTML

### Screenie
Screen Shot 2020-11-12 at 11 33 58 AM

### How to 🎩

* Chroma should be 💯

### CODE REVIEW

File: src/components/Filters/Filters.tsx [hunk 1]
```diff
@@ -187,7 +187,7 @@ class FiltersInner extends Component<CombinedProps, State> {
             aria-expanded={filterIsOpen}
           >
             <div className={styles.FilterTriggerLabelContainer}>
-              <h2 className={styles.FilterTriggerTitle}>
+              <h3 className={styles.FilterTriggerTitle}>
```
Comment (alex-page) [inline src/components/Filters/Filters.tsx:190]: Are they always going to follow a h2? Is this known?

Comment (AndrewMusgrave) [reply src/components/Filters/Filters.tsx:190]: They should. The filters content opens into a sheet (portal), so app adjacent. If we make the sheet title an `h2`. Making the filter headers `h3` should remain semantic 👌

Comment (alex-page) [reply src/components/Filters/Filters.tsx:190]: Interesting, I wonder if there is an example where they would be used in another context. That makes sense though.

Comment (AndrewMusgrave) [reply src/components/Filters/Filters.tsx:190]: Nope! The filters components render all of the markup, so unless it changes in Polaris that'll always be the case 😄

**Review by kyledurand — APPROVED** [review]
Looks good to me

**Review by ashwinnarayanan2001 — APPROVED** [review]
Looking good.

**Review by emmaorhun — APPROVED** [review]
👌

**Pull request merged**